cmd/push.go

```diff
@@ -158,52 +166,58 @@ func runPush(dev *model.Dev, autoDeploy bool, imageTag, oktetoRegistryURL, progr
 		return err
 	}
 
-	buildKitHost, isOktetoCluster, err := build.GetBuildKitHost()
+	imageTag, err = buildImage(dev, imageTag, imageFromDeployment, oktetoRegistryURL, noCache, progress)
 	if err != nil {
 		return err
 	}
 
-	imageTag = build.GetDevImageTag(dev, imageTag, imageFromDeployment, oktetoRegistryURL)
-	log.Infof("pushing with image tag %s", imageTag)
-
-	var imageDigest string
-	buildArgs := model.SerializeBuildArgs(dev.Push.Args)
-	imageDigest, err = build.Run(buildKitHost, isOktetoCluster, dev.Push.Context, dev.Push.Dockerfile, imageTag, dev.Push.Target, noCache, buildArgs, progress)
-	if err != nil {
-		return fmt.Errorf("error building image '%s': %s", imageTag, err)
-	}
-	if imageDigest != "" {
-		imageWithoutTag := build.GetRepoNameWithoutTag(imageTag)
-		imageTag = fmt.Sprintf("%s@%s", imageWithoutTag, imageDigest)
-	}
-
 	spinner := utils.NewSpinner(fmt.Sprintf("Pushing source code to the development environment '%s'...", dev.Name))
 	spinner.Start()
 	defer spinner.Stop()
+
 	if create {
```

Comment:
Is this recreating the service?
I think it is not. It could work if you check the deployment autocreate annotation instead of the `create` envvar.